File: storagemanager/src/main/java/com/example/alexeyglushkov/cachemanager/clients/CacheStatus.java
package com.example.alexeyglushkov.cachemanager.clients;

import androidx.annotation.Nullable;

/**
 * Created by alexeyglushkov on 03.03.18.
 */

public class CacheStatus {

    public static boolean canLoadFromCache(@Nullable Cache cache) {
        if (cache == null) {
            return false;
        }

        Cache.CacheMode cacheMode = cache.getCacheMode();
        return cacheMode != Cache.CacheMode.IGNORE_CACHE &&
                cacheMode != Cache.CacheMode.ONLY_STORE_TO_CACHE;
    }

    public static boolean canWriteToCache(@Nullable Cache client) {
        if (client == null) {
            return false;
        }

        Cache.CacheMode cacheMode = client.getCacheMode();
        return cacheMode != Cache.CacheMode.IGNORE_CACHE &&
                cacheMode != Cache.CacheMode.ONLY_LOAD_FROM_CACHE;
    }
}

File: storagemanager/src/main/java/com/example/alexeyglushkov/cachemanager/clients/CacheStatus.kt
package com.example.alexeyglushkov.cachemanager.clients

import com.example.alexeyglushkov.cachemanager.clients.Cache.CacheMode

/**
 * Created by alexeyglushkov on 03.03.18.
 */
object CacheStatus {
    fun canLoadFromCache(cache: Cache?): Boolean {
        if (cache == null) {
            return false
        }

        val cacheMode = cache.cacheMode
        return cacheMode != CacheMode.IGNORE_CACHE &&
                cacheMode != CacheMode.ONLY_STORE_TO_CACHE
    }

    fun canWriteToCache(client: Cache?): Boolean {
        if (client == null) {
            return false
        }

        val cacheMode = client.cacheMode
        return cacheMode != CacheMode.IGNORE_CACHE &&
                cacheMode != CacheMode.ONLY_LOAD_FROM_CACHE
    }
}